### PR TITLE
 feat(TDI-41396): add additional parameters field

### PR DESCRIPTION
**Pull request by Voievidko**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-41396

**What is the new behavior?**
table field in additional settings was added

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.bigdata/components/tCosmosDBConnection/tCosmosDBConnection_begin.javajet [hunk 1]
```diff
@@ -33,35 +36,147 @@
     String authDB = ElementParameterParser.getValue(node, "__AUTHENTICATION_DATABASE__");
     String usedAuthenticationDB = useAuthDB ? authDB : dbname;
     boolean queryOptionNoTimeOut = "true".equalsIgnoreCase(ElementParameterParser.getValue(node, "__QUERYOPTION_NOTIMEOUT__"));
+    boolean applyOptionalAttributes = "true".equals(ElementParameterParser.getValue(node, "__APPLY_OPTIONAL_ATTRIBUTES__"));
     
     String mongoDbDriver = "com.mongodb.MongoClient";
     
     boolean isLog4jEnabled = ("true").equals(ElementParameterParser.getValue(node.getProcess(), "__LOG4J_ACTIVATE__"));
 
+    List<String> intAttrisList = Arrays.asList(
```
Comment (mbasiuk-talend) [inline main/plugins/org.talend.designer.components.bigdata/components/tCosmosDBConnection/tCosmosDBConnection_begin.javajet]: What is the **attris** ? could you please use more correct naming.

File: main/plugins/org.talend.designer.components.bigdata/components/tCosmosDBConnection/tCosmosDBConnection_begin.javajet [hunk 2]
```diff
@@ -33,35 +36,147 @@
...
-        .build();
+                    .socketFactory(javax.net.ssl.SSLSocketFactory.getDefault())
         <%
     }
+%>
+
+                    .build();
+<%
 
+    // Client Credentials
```
Comment (mbasiuk-talend) [inline main/plugins/org.talend.designer.components.bigdata/components/tCosmosDBConnection/tCosmosDBConnection_begin.javajet]: Why this comment is added here? Is this give an additional information to comment belove?

File: main/plugins/org.talend.designer.components.bigdata/components/tCosmosDBConnection/tCosmosDBConnection_java.xml [hunk 3]
```diff
@@ -102,6 +102,48 @@
...
+                  <ITEM NAME="MAXCONNECTIONLIFETIME" VALUE="MAXCONNECTIONLIFETIME"/>
+                  <ITEM NAME="MAXWAITTIME" VALUE="MAXWAITTIME"/>
+                  <ITEM NAME="MINCONNECTIONSPERHOST" VALUE="MINCONNECTIONSPERHOST"/>
+                  <ITEM NAME="MINHEARTBEATFREQUENCY" VALUE="MINHEARTBEATFREQUENCY"/>
+                  <ITEM NAME="REQUIREDREPLICASETNAME" VALUE="REQUIREDREPLICASETNAME"/>
+                  <ITEM NAME="SERVERSELECTIONTIMEOUT" VALUE="SERVERSELECTIONTIMEOUT"/>
+                  <ITEM NAME="SOCKETKEEPALIVE" VALUE="SOCKETKEEPALIVE"/>
+                  <ITEM NAME="SOCKETTIMEOUT" VALUE="SOCKETTIMEOUT"/>
+                  <ITEM NAME="SSLINVALIDHOSTNAMEALLOWED" VALUE="SSLINVALIDHOSTNAMEALLOWED"/>
+                  <ITEM NAME="THREADSALLOWEDTOBLOCKFORCONNECTIONMULTIPLIER" VALUE="THREADSALLOWEDTOBLOCKFORCONNECTIONMULTIPLIER"/>
```
Comment (mbasiuk-talend) [inline main/plugins/org.talend.designer.components.bigdata/components/tCosmosDBConnection/tCosmosDBConnection_java.xml]: My bleeding eyes...

Comment (Voievidko) [reply main/plugins/org.talend.designer.components.bigdata/components/tCosmosDBConnection/tCosmosDBConnection_java.xml]: All properties was renamed

File: main/plugins/org.talend.designer.components.bigdata/components/tCosmosDBConnection/tCosmosDBConnection_begin.javajet (hunk 1 above)
Comment (Voievidko) [reply main/plugins/org.talend.designer.components.bigdata/components/tCosmosDBConnection/tCosmosDBConnection_begin.javajet]: I did better naming

**Pull request merged**